java_gen/pre-written/src/main/java/org/projectfloodlight/openflow/types/IPv4Address.java

```diff
@@ -117,6 +117,61 @@ public boolean isMulticast() {
         return ((rawValue >>> 24) & 0xF0) == 0xE0;
     }
 
+    public enum IPAddressType {
+        THIS_NETWORK,
+        PRIVATE_NETWORK,
+        PUBLIC_DATA_NETWORK,
+        CABLE_TV_NETWORK,
+        INTERNET_HOST_LOOPBACK_NETWORK,
+        LINK_LOCAL_NETWORK,
+        TEST_NET,
+        RELAY_ANYCAST_NETWORK,
+        INTERCONNECT_NETWORK,
+        MULTICAST_NETWORK,
+        RESERVED_NETWORK,
+        HOST_ADDRESS
```

Comment:
Are we sure that `HOST_ADDRESS` is a valid terminology here? IMHO, a valid `HostAddress` depends on the network in which it exists; in particular it must be within the network and not the network or broadcast address of that network.